### PR TITLE
machines: clone breaks for non-postgres apps

### PR DESCRIPTION
**Pull request by ignoramous**
More: https://community.fly.io/t/7265

Exec:

```bash
➜  flyctl ✗ go build -o fly2

➜  flyctl ✗ go test
?   	github.com/superfly/flyctl	[no test files]

➜ fly2 m clone d5683061b4158e --config fly.machines.toml --name udns-gru2 --region gru
Cloning machine in region gru for app udns
Machine d5683061b4158e was cloned to 73287111be1385
```

### CODE REVIEW

File: flaps/flaps.go [hunk 1]
```diff
@@ -114,6 +114,17 @@ func (f *Client) Start(ctx context.Context, machineID string) (*api.MachineStart
 	return out, nil
 }
 
+func (f *Client) TryWait(ctx context.Context, machine *api.Machine, state string, tries int) (err error) {
```
Comment (davissp14) [inline flaps/flaps.go:117]: Were you experiencing a specific issue that resulted in this change?

File: internal/command/machine/clone.go [hunk 2]
```diff
@@ -96,7 +96,7 @@ func runMachineClone(ctx context.Context) (err error) {
 	// This is a temperary hack to add volume support for PG apps.
 	// Flaps does not currently specify the volume name within the Machine mount spec,
 	// which is required before we can handle this more generally.
-	if app.PostgresAppRole.Name == "postgres_cluster" {
+	if app.PostgresAppRole != nil && app.PostgresAppRole.Name == "postgres_cluster" {
```
Comment (davissp14) [inline internal/command/machine/clone.go:99]: 👍

**Review by davissp14 — CHANGES_REQUESTED** [review]
Overall looks good!  Thank you!  Just looking for some context on the `TryWait` addition.

File: flaps/flaps.go (hunk 1 above)
Comment (ignoramous) [reply flaps/flaps.go:117]: Yes, in one of the tests, machine took more than `30s` to come up.

Comment (ignoramous) [reply flaps/flaps.go:117]: `run.go` anticipates it, too: https://github.com/superfly/flyctl/blob/7fe7fd5b4d0f21e18e950a344675b32b3978b447/internal/command/machine/run.go#L298 (may be `update` should use this, instead).

Comment (ignoramous) [reply flaps/flaps.go]: Done

**Review by davissp14 — APPROVED** [review]
Thank you!

**Pull request merged**